connect/runtime/src/main/java/org/apache/kafka/connect/runtime/SourceTaskOffsetCommitter.java

```diff
@@ -105,6 +105,11 @@ public void remove(ConnectorTaskId id) {
     }
 
     private void commit(WorkerSourceTask workerTask) {
+        if (!workerTask.shouldCommitOffsets()) {
```

Comment:
I initially wanted to do just that, but held off for two reasons:

1. The `SourceTaskOffsetCommitter` log message may be valuable to some users, especially if they've muted the `WorkerSourceTask` namespace or at least raised its level to `INFO`.
2. It's a little easier to test this logic separately by creating a separate method that can be queried on its own. The `boolean` return value of `WorkerSourceTask::commitOffsets` makes it difficult to determine if an offset commit has been attempted and failed, or skipped entirely.

I'm not super attached to either approach; if you have thoughts on why one or the other is better I'm happy to hear them.

Reply:
I prefer the encapsulation approach but will leave the decision to you as it's a minor nitpick :).

I prefer encapsulation because the caller to `commitOffsets` need not worry about first calling `shouldCommitOffsets` and only if it's true then I should call `commitOffsets`.  Think like we don't need to do  `!ArrayList.isEmpty` check before iterating over it :).

Reply:
So a few things here:

1. Even if `commitOffsets` is called without the `shouldCommitOffsets` check, if any of the conditions that lead to `shouldCommitOffsets` returning `false` arise in the middle of that call to `commitOffsets`, the offset commit will be prematurely aborted and, in some cases, effectively a no-op. So with the analogy of iterating over a list--this is actually perfectly in line with that example; you don't _have_ to call `List::isEmpty` before iterating since iterating over an empty list is effectively a no-op, but you can choose to if you'd like.
2. We could add a `shouldCommitOffsets` guard against the end-of-life offset commit, but since that only occurs once per task and, as mentioned above, it'll now effectively be a no-op if offset commit isn't possible, I don't think it's really necessary. The points of this PR are to avoid cluttering up the single-threaded source task offset committer and to avoid continually emitting useless and confusing log messages. Gating on a one-time offset commit that only takes place once the task is shut down isn't really necessary for either of these.
3. The `SourceTaskOffsetCommitter::commit` method's "{} Committing offsets" log message is super old and I'm really hesitant to mess with that, even by moving it into a separate class. I know this seems trivial to developers but logs are often the first or second line of defense users have in diagnosing incorrect or unexpected behavior on their workers and I don't want to make that any harder, especially for users with log processing pipelines that parse logging namespaces from messages and allow users to search based on that information.

Reply:
makes sense